### PR TITLE
[3.0] bugfix: spriteframe support flip X & flip Y

### PR DESCRIPTION
**Pull request by PatriceJiang**
Re: https://github.com/cocos-creator/3d-tasks/issues/5119

Changes:
 *

### CODE REVIEW

File: cocos/core/assets/sprite-frame.ts [hunk 1]
```diff
@@ -479,11 +504,13 @@ export class SpriteFrame extends Asset {
...
 
-    protected _atlasUuid: string = '';
+    protected _atlasUuid = '';
     // @ts-expect-error
     protected _texture: TextureBase | RenderTexture;
 
-    protected _flipUv = false;
+    protected _flipUvY = false;
+
+    protected _flipUvX = false;
```
Comment (pandamicro) [inline cocos/core/assets/sprite-frame.ts]: Please make the name same as public property

File: cocos/core/assets/sprite-frame.ts [hunk 2]
```diff
@@ -1019,6 +1158,34 @@ export class SpriteFrame extends Asset {
...
+            nv: v.nv?.slice(0),
+            v: v.v?.slice(0),
+        } : null as any;
+        sp.uv = this.uv.slice(0);
+        sp.uvHash = this.uvHash;
+        sp.unbiasUV = this.unbiasUV.slice(0);
+        sp.uvSliced = this.uvSliced.slice(0);
+        sp._rect = this._rect.clone();
+        sp._offset = this._offset.clone();
+        sp._originalSize = this._originalSize.clone();
```
Comment (pandamicro) [inline cocos/core/assets/sprite-frame.ts]: These properties are all created with new SpriteFrame, the clone is overwriting them with new objects, which create garbage objects.
I prefer they are calculated with invocations like `_calculateUV` and `reset` or `sp.rect = this._rect;`

**Review by pandamicro — CHANGES_REQUESTED** [review]

File: cocos/core/assets/sprite-frame.ts (hunk 1 above)
Comment (PatriceJiang) [reply cocos/core/assets/sprite-frame.ts]: Okay

File: cocos/core/assets/sprite-frame.ts (hunk 2 above)
Comment (PatriceJiang) [reply cocos/core/assets/sprite-frame.ts]: These 'garbage objects' are reused in the next change.

**Review by pandamicro — APPROVED** [review]

**Pull request merged**